### PR TITLE
add rustfmt.toml and use nightly formatting

### PR DESCRIPTION
**Pull request by mattsse**
Add `rustfmt.toml` with some configs, listed here [https://rust-lang.github.io/rustfmt/](https://rust-lang.github.io/rustfmt/)

use nightly formatting `cargo +nightly fmt`

add separate CI task for lining

1 clippy error, could either fix here or separately

### CODE REVIEW

File: .github/workflows/rust.yml
Comment (ralexstokes) [inline .github/workflows/rust.yml:50]: should we add this to the `build` job?

File: .github/workflows/rust.yml
Comment (ralexstokes) [inline .github/workflows/rust.yml:61]: I'm fine to drop `--verbose` although I see that `--all` is a deprecated option?

I think we can use `--all-targets` and get the same effect?

File: mev-boost-rs/src/relay_mux.rs
Comment (ralexstokes) [inline mev-boost-rs/src/relay_mux.rs:17]: ah the "auto-grouping" of imports is very nice

**Review by ralexstokes — COMMENTED** [review]
nice work!

I like breaking out the linting into a separate CI job.

I tend to prefer just the "stock" configuration for some tool to reduce the amount of overhead needed to operate a given codebase, e.g. requiring `rustfmt.toml` here although I'm open to these changes

I left some questions in the comments to address before merging. And about the CI failures, I'm fine to merge this in now if you want to follow up soon with another PR fixing the lints -- from what I can see we can just add some `Box`es to the offending error variants and satisfy the lint

File: .github/workflows/rust.yml
Comment (mattsse) [reply .github/workflows/rust.yml:61]: right, also no needed since we don't have non default members

**Comment by mattsse** [comment]
made changes to the action


most of the rustfmt.toml settings concern line, comment wrapping, and removing trailing commas semicolons
some are a bit more opinionated, but imo really only

https://rust-lang.github.io/rustfmt/?version=v1.5.1&search=#binop_separator

https://rust-lang.github.io/rustfmt/?version=v1.5.1&search=#use_field_init_shorthand

and 

https://rust-lang.github.io/rustfmt/?version=v1.5.1&search=#use_small_heuristics

which allows more width for calls etc before it wraps

but happy to use the defaults instead

File: .github/workflows/rust.yml
Comment (mattsse) [reply .github/workflows/rust.yml:52]: done

**Comment by ralexstokes** [comment]
im fine w/ the config you added, can you follow up with a PR to fix the lint?

**Comment by mattsse** [comment]
fixed by boxing the large error variant

**Review by ralexstokes — APPROVED** [review]
excellent, thanks for this!

**Pull request merged**
